tests/test_solving.py

```diff
@@ -230,3 +230,55 @@ def dep3(
 
     container = Container()
     container.solve(Dependant(dep3, scope=None), scopes=[None])
+
+
+def test_infer_scope_1() -> None:
+    # This dep must be inferred into the "app" scope
+    # because that's the only scope we enter
+    # Otherwise we'd get an error
+
+    def dep() -> Iterator[None]:
+        yield None
+
+    container = Container()
+    solved = container.solve(Dependant(dep), scopes=["app"])
+    with container.enter_scope("app") as state:
+        container.execute_sync(solved, SyncExecutor(), state=state)
+
+
+def test_infer_scope_2() -> None:
+    # Since None is a known scope, we don't change it
```

Comment:
This is mainly for backwards compatibility. I think it would be cleaner to say "`None` is not a valid scope, it's just a placeholder for unset / undefined scope". Might make that breaking change down the road.